File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<int>> Map;
        for(int i = 0; i < strs.size(); ++i) {
            string s = strs[i];
            sort(s.begin(), s.end());
            Map[s].push_back(i);
        }
        vector<vector<string>> ans;
        for(auto [k, v] : Map) {
            ans.push_back({});
            for(int i : v) {
                ans.back().push_back(strs[i]);
            }
        }
        return ans;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dict = {}
        for x in strs:
            xSorted = sorted(x)
            key = "".join(xSorted)
            if key in dict:
                dict[key].append(x)
            else:
                
                dict[key] = [x]
                
        return dict.values()